querqy-core/src/main/java/querqy/rewrite/lookup/TrieMapStateExchangingCollector.java

```diff
@@ -0,0 +1,180 @@
+package querqy.rewrite.lookup;
+
+import querqy.CompoundCharSequence;
+import querqy.model.Term;
+import querqy.rewrite.commonrules.model.TermMatch;
+import querqy.rewrite.commonrules.model.TermMatches;
+import querqy.rewrite.lookup.model.Match;
+import querqy.rewrite.lookup.model.Sequence;
+import querqy.trie.State;
+import querqy.trie.States;
+import querqy.trie.TrieMap;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+public class TrieMapStateExchangingCollector<T> implements StateExchangingCollector<State<T>, T> {
+
+    private final TrieMap<T> trieMap;
+    private final boolean ignoreCase;
+
+    private final List<Match<T>> matches = new ArrayList<>();
+
+    private TrieMapStateExchangingCollector(final TrieMap<T> trieMap, final boolean ignoreCase) {
+        this.trieMap = trieMap;
+        this.ignoreCase = ignoreCase;
+    }
+
+    @Override
+    public Optional<State<T>> evaluateTerm(final Term term) {
+        // TODO: why with field?
```

Comment:
This left open the possibility to specify fields in the input declaration.

Apart from this, in our object model, Terms are different depending on their fields. If we did not consider field here, we'd have to add a special Term type to our model that does not have a field (strictly spoken ;-) )